modules/tinymce/src/core/test/ts/browser/fmt/ListItemFormatTest.ts

```diff
@@ -0,0 +1,336 @@
+          selection: { startPath: [ 0, 0, 0 ], soffset: 0, finishPath: [ 0, 0, 0 ], foffset: 3 },
+          expected: '<ul><li style="font-size: 30px;"><span style="font-size: 30px;">abc</span></li></ul>'
+        })
+      );
+
+      it('TINY-8961: applying font family to the entire contents of a LI should also apply that font family to the LI', () =>
+        testApplyInlineListFormat({
+          format: 'fontname',
+          value: 'Arial',
+          input: '<ul><li>abc</li></ul>',
+          selection: { startPath: [ 0, 0, 0 ], soffset: 0, finishPath: [ 0, 0, 0 ], foffset: 3 },
+          expected: '<ul><li style="font-family: Arial;"><span style="font-family: Arial;">abc</span></li></ul>'
+        })
+      );
+
+      it('TINY-8961: applying bold to a fully selected LI with deep nested content should apply bold to the LI', () =>
+        testApplyInlineListFormat({
+          format: 'bold',
+          input: '<ul><li><em><span id="x">abc</span></em></li></ul>',
+          selection: { startPath: [ 0, 0, 0, 0, 0 ], soffset: 0, finishPath: [ 0, 0, 0, 0, 0 ], foffset: 3 },
+          expected: '<ul><li style="font-weight: bold;"><strong><em><span id="x">abc</span></em></strong></li></ul>',
+        })
+      );
+
+      it('TINY-8961: applying bold to a partially selected start of a LI should not apply bold to the LI', () =>
+        testApplyInlineListFormat({
+          format: 'bold',
+          input: '<ul><li>abc</li></ul>',
+          selection: { startPath: [ 0, 0, 0 ], soffset: 0, finishPath: [ 0, 0, 0 ], foffset: 2 },
+          expected: '<ul><li><strong>ab</strong>c</li></ul>'
+        })
+      );
+
+      it('TINY-8961: applying bold to a partially selected end of a LI should not apply bold to the LI', () =>
+        testApplyInlineListFormat({
+          format: 'bold',
+          input: '<ul><li>abc</li></ul>',
+          selection: { startPath: [ 0, 0, 0 ], soffset: 1, finishPath: [ 0, 0, 0 ], foffset: 3 },
+          expected: '<ul><li>a<strong>bc</strong></li></ul>'
+        })
+      );
+
+      it('TINY-8961: applying bold to a partially selected LI should not apply bold to the LI', () =>
+        testApplyInlineListFormat({
+          format: 'bold',
+          input: '<ul><li>abc</li></ul>',
+          selection: { startPath: [ 0, 0, 0 ], soffset: 1, finishPath: [ 0, 0, 0 ], foffset: 2 },
+          expected: '<ul><li>a<strong>b</strong>c</li></ul>'
+        })
+      );
+
+      it('TINY-8961: applying underline to the entire contents of a LI should not apply that to the LI since it will not be rendered', () =>
+        testApplyInlineListFormat({
+          format: 'underline',
+          input: '<ul><li>abc</li></ul>',
+          selection: { startPath: [ 0, 0, 0 ], soffset: 0, finishPath: [ 0, 0, 0 ], foffset: 3 },
+          expected: '<ul><li><span style="text-decoration: underline;">abc</span></li></ul>'
+        })
+      );
+    });
+
+    context('Apply inline formats to multiple list elements', () => {
```

Comment:
at different from the previous `context`, the tests descriptions here are missing on what the style is applied for example: `should also apply that bold to the LI`, could it make sense to specify it also in these tests or remove it from the previous tests?

Reply:
Good point the labels where very inconsistent and also hard to understand since they where lacking context. I switched the `lis` `li` and `list elements` to be all `LIs` or `LI` since the `list item` and `list element` is pretty verbose.